### PR TITLE
Earizon pr85 review

### PR DESCRIPTION
**Pull request by adrianhopebailie**
"Bugs" found during testing with java connector

Reopened from #89 to target`master`

### CODE REVIEW

File: crypto-conditions/src/main/java/org/interledger/cryptoconditions/PreimageSha256Fulfillment.java [hunk 1]
```diff
@@ -24,7 +24,7 @@
    */
   static PreimageSha256Fulfillment from(final byte[] preimage) {
     Objects.requireNonNull(preimage);
-    final String encodedPreimage = Base64.getUrlEncoder().encodeToString(preimage);
+    final String encodedPreimage = Base64.getEncoder().encodeToString(preimage);
```
Comment (adrianhopebailie) [inline crypto-conditions/src/main/java/org/interledger/cryptoconditions/PreimageSha256Fulfillment.java]: As discussed in #89 I think we should revert this.

If an external application needs the Base64 (not Bse64Url) encoded form it should convert between them

File: ilp-core/src/main/java/org/interledger/core/InterledgerAddress.java [hunk 2]
```diff
@@ -35,7 +35,7 @@
 public interface InterledgerAddress {
 
   String REGEX = "(?=^.{1,1023}$)"
-      + "^(g|private|example|peer|self|test[1-3])[.]([a-zA-Z0-9_~-]+[.])*([a-zA-Z0-9_~-]+)?$";
+      + "^(g|private|example|peer|self|test([1-3])?)[.]([a-zA-Z0-9_~-]+[.])*([a-zA-Z0-9_~-]+)?$";
```
Comment (adrianhopebailie) [inline ilp-core/src/main/java/org/interledger/core/InterledgerAddress.java]: We can revert this too as it's addressed in #90

File: ilp-core/src/main/java/org/interledger/core/InterledgerProtocolException.java [hunk 3]
```diff
@@ -27,4 +27,10 @@ public InterledgerProtocolException(final InterledgerRejectPacket interledgerRej
   public InterledgerRejectPacket getInterledgerRejectPacket() {
     return interledgerRejectPacket;
   }
+
+  @Override
+  public String toString(){
+      return interledgerRejectPacket.toString();
+  }
+
 }
```
Comment (adrianhopebailie) [inline ilp-core/src/main/java/org/interledger/core/InterledgerProtocolException.java]: As discussed in #89 I think this should include the value of `this.getMessage()`

File: crypto-conditions/src/main/java/org/interledger/cryptoconditions/PreimageSha256Fulfillment.java (hunk 1 above)
Comment (sappenin) [reply crypto-conditions/src/main/java/org/interledger/cryptoconditions/PreimageSha256Fulfillment.java]: Agree. Let's revert for now and discuss in #100.

**Comment by sappenin** [comment]
Closing this for now - as discussed on the last call, there's only one change in here that we actually want to keep, but it probably makes sense to open a new PR for that. I've opened #134 to track that and assigned it to @earizon.

**Pull request closed without merging**